docs/src/test/java/org/elasticsearch/smoketest/DocsClientYamlTestSuiteIT.java

```diff
@@ -64,5 +90,117 @@ protected ClientYamlTestClient initClientYamlTestClient(ClientYamlSuiteRestSpec
                                                             List<HttpHost> hosts, Version esVersion) throws IOException {
         return new ClientYamlDocsTestClient(restSpec, restClient, hosts, esVersion);
     }
-}
 
+    /**
+     * Compares the the results of running two analyzers against many random
+     * strings. The goal is to figure out if two anlayzers are "the same" by
+     * comparing their results. This is far from perfect but should be fairly
+     * accurate, especially for gross things like missing {@code decimal_digit}
+     * token filters, and should be fairly fast because it compares a fairly
+     * small number of tokens.
+     */
+    private static class CompareAnalyzers implements ExecutableSection {
+        private static ConstructingObjectParser<CompareAnalyzers, XContentLocation> PARSER =
+            new ConstructingObjectParser<>("test_analyzer", false, (a, location) -> {
+                String index = (String) a[0];
+                String first = (String) a[1];
+                String second = (String) a[2];
+                return new CompareAnalyzers(location, index, first, second);
+            });
+        static {
+            PARSER.declareString(constructorArg(), new ParseField("index"));
+            PARSER.declareString(constructorArg(), new ParseField("first"));
+            PARSER.declareString(constructorArg(), new ParseField("second"));
+        }
+        private static CompareAnalyzers parse(XContentParser parser) throws IOException {
+            XContentLocation location = parser.getTokenLocation();
+            CompareAnalyzers section = PARSER.parse(parser, location);
+            assert parser.currentToken() == Token.END_OBJECT;
```

Comment:
Can you add a message here so it'll be helpful if someone accidentally misses a closing token?